python/lsst/cp/pipe/cpCombine.py

```diff
@@ -343,6 +350,10 @@ def run(self, inputExps, inputScales=None, inputDims=None):
         self.combineHeaders(inputExps, combinedExp,
                             calibType=self.config.calibrationType, scales=expScales)
 
+        # Set the detector
+        inputDetector = inputExps[0].getDetector()
```

Comment:
I think this calculation should be moved to the earlier logic where you build the list and work out which one is not-none.